src/boot/efi/meson.build

```diff
@@ -39,312 +146,206 @@ systemd_boot_sources = '''
 '''.split()
 
 stub_sources = '''
+        cpio.c
         initrd.c
         splash.c
         stub.c
-        cpio.c
 '''.split()
 
-if efi_arch in ['x86', 'x86_64']
+if efi_arch[1] in ['ia32', 'x86_64']
         stub_sources += 'linux_x86.c'
 else
         stub_sources += 'linux.c'
 endif
 
-if conf.get('ENABLE_EFI') == 1 and get_option('gnu-efi') != 'false'
-        efi_cc = get_option('efi-cc')
-        if efi_cc.length() == 0
-                efi_cc = cc.cmd_array()
+conf.set10('HAVE_GNU_EFI', true)
+conf.set_quoted('EFI_MACHINE_TYPE_NAME', efi_arch[0])
+
+efi_conf = configuration_data()
+efi_conf.set_quoted('EFI_MACHINE_TYPE_NAME', efi_arch[0])
+efi_conf.set10('ENABLE_TPM', get_option('tpm'))
+
+foreach ctype : ['color-normal', 'color-entry', 'color-highlight', 'color-edit']
+        c = get_option('efi-' + ctype).split(',')
+        efi_conf.set(ctype.underscorify().to_upper(), 'EFI_TEXT_ATTR(@0@, @1@)'.format(
```

Comment:
Previously, this was `(A | B)`, but now `EFI_TEXT_ATTR(A, B)`. Is this intended?

Reply:
Yes, this is desired. It allows the technically reserved 8th bit to be used for color too, should the efi support it.